### ISSUE TITLE
Add targetProperty for parameter grouper

### ISSUE THREAD

**Issue by iscai-msft** [Azure/autorest.modelerfour#165]
**Describe the issue**
It would be nice to add a targetProperty in the yaml so grouped parameters can access. @lmazuel suggested ref3 having a targetProperty to the property in the group shown below
![MicrosoftTeams-image](https://user-images.githubusercontent.com/43154838/74392573-7cba5c80-4dbc-11ea-99e2-f83b41bb2c31.png)

**Comment by fearthecowboy** [comment]
@iscai-msft Just trying to understand what you're looking for here - you want the groupSchema to point to the things that it's grouping together?

Because group schemas can span multiple operations, I believe that each groupProperty in the groupSchema already has an 'originalParameter' array which points to the parameters that it was created to replace.

**Comment by pakrym** [comment]
I would love this too. Having a reference to a property that grouped parameter is mapped to would make things way easier to generate.

**Comment by fearthecowboy** [comment]
The `GroupSchema ` does provide this now --

`GroupSchema` properties each have an `originalParameter` collection which is the set of parameters that property is replacing (most of the time, it'll be a single parameter in the array, but it multiple operations use the same parameter group, and each operation lists the same named parameter, the `GroupParameter` will have to list both. (since the schema is shared among multiple operations)

**Comment by fearthecowboy** [comment]
Example https://github.com/Azure/autorest.modelerfour/blob/master/modelerfour/test/scenarios/azure-special-properties/grouped.yaml#L122

**Comment by pakrym** [comment]
Yeah, I understand that there is a way to find it but it requires enumerating all properties and looking for one that has correct item in it's collection. Instead of just having a direct reference.

**Comment by fearthecowboy** [comment]
I see.

I can replaced grouped with a structure that gives you all that information:
``` ts
 /** if the parameter gets grouped into a parameter group this points to the details */
    groupedBy?: { 
      /** the parameter that replaces this in this operation/request */
      parameter: Parameter,

      /** the schema that contains the property that is created */
      schema: GroupSchema,
      
      /** the property in the schema that represents this parameter */
      property: GroupProperty
    }
```

so instead of `groupedBy` pointing to the parameter alone, `groupedBy.parameter` would  and `groupedBy.schema` and `groupedBy.property` would tell you where it went

Would that suffice?

**Comment by pakrym** [comment]
Yeah, this would work for me. @iscai-msft what about you?

**Comment by iscai-msft** [comment]
yep that looks great, thanks @fearthecowboy
